src/api/argo-templates/base/sat-general-template-iuf.argo.yaml

```diff
@@ -0,0 +1,146 @@
+              template.
+      outputs:
+        parameters:
+          - name: script_stdout
+            valueFrom:
+              path: "{{inputs.parameters.script_stdout_file}}"
+      script:
+        image: artifactory.algol60.net/sat-docker/stable/cray-sat:3.20.0
+        command: [sh]
+        source: |
+          #!/bin/sh
+          ts="$(mktemp)"
+          cat <<'SAT_GENERAL_EOF' > "${ts}"
+          function setup() {
+            set -e
+            sat init
+
+            # Write the auth token parameter to the file
+            JSON_FMT='{"access_token":"%s"}\n'
+            printf "$JSON_FMT" "{{inputs.parameters.auth_token}}" > /root/.config/sat/token.json
+            echo -n $(tr -d "\n\r" < /root/.config/sat/token.json) > /root/.config/sat/token.json
+
+            # Set up the SAT config to use the token file
+            sed -i -e 's|.*token_file.*|token_file = "/root/.config/sat/token.json"|' /root/.config/sat/sat.toml
+
+            # TODO: Fix sat bootprep code to load kube config from cluster
+            # To workaround this issue for now, construct kubeconfig file in the pod
+            secret=`kubectl -n argo get serviceaccount/default -o jsonpath='{.secrets[0].name}'`
+            token=`kubectl -n argo get secret $secret -o jsonpath='{.data.token}'| base64 -d`
+            mkdir -p mykubeconfig
+            cat << EOF > mykubeconfig/admin.conf
+            apiVersion: v1
+            kind: Config
+            current-context: default
+            contexts:
+              - context:
+                  cluster: kubernetes
+                  user: default
+                  namespace: default
+                name: default
+            clusters:
+              - cluster:
+                  server: https://kubeapi-vip.local:6442
+                  insecure-skip-tls-verify: true
+                name: kubernetes
+            users:
+            - name: default
+              user:
+                token: ${token}
+          EOF
+            export KUBECONFIG=mykubeconfig/admin.conf
+
+            # TODO: This is required to build images in IMS with sat bootprep. Find a way to pass
+            # an IMS public key id into the argo workflow. Perhaps from the IUF CLI.
+            ssh-keygen -f ~/.ssh/id_rsa -N ''
+            set +e
+          }
+
+          function execute_script_content() {
+            {{inputs.parameters.script_content}}
+          }
+
+          function main() {
+            setup
+            execute_script_content | tee {{inputs.parameters.script_stdout_file}}
+          }
+
+          main
+          SAT_GENERAL_EOF
+
+          bash ${ts}
+        env:
+          - name: REQUESTS_CA_BUNDLE
+            value: "{{inputs.parameters.ca_certs_bundle}}"
+        volumeMounts:
+          - name: certs # mount cluster certs to ca-certificates.crt for curl/http libraries
+            mountPath: "{{inputs.parameters.ca_certs_bundle}}"
+            subPath: platform-ca-certs.crt
```

Comment:
Running this on the master nodes results in leaving behind orphaned pod state in /var/lib/kubelet because it is not able to unmount the subPath mounts.  This is caused by the way /var/lib/kubelet is mounted on the master nodes.  See CASMTRIAGE-4827.

Over time this causes a lot of orphaned state to build up and kubelet becomes very bogged down trying to clean them up.

Reply:
I'm working on an alternative to using this `subPath` today, and I'll push a fix ASAP.

Reply:
This has been addressed.